### PR TITLE
Process delimiters on the object info cache hit

### LINKED ISSUE #611: Fix delimeter issues
https://github.com/nspcc-dev/neofs-s3-gw/commit/42a761c1f25c046b019e063c7e71a52afb8c7086 broke these delimeter tests:

```
s3tests_boto3.functional.test_s3.test_bucket_list_delimiter_basic
s3tests_boto3.functional.test_s3.test_bucket_listv2_delimiter_basic
s3tests_boto3.functional.test_s3.test_bucket_listv2_encoding_basic
s3tests_boto3.functional.test_s3.test_bucket_list_encoding_basic
s3tests_boto3.functional.test_s3.test_bucket_list_delimiter_alt
s3tests_boto3.functional.test_s3.test_bucket_listv2_delimiter_alt
s3tests_boto3.functional.test_s3.test_bucket_list_delimiter_percentage
s3tests_boto3.functional.test_s3.test_bucket_listv2_delimiter_percentage
s3tests_boto3.functional.test_s3.test_bucket_list_delimiter_whitespace
s3tests_boto3.functional.test_s3.test_bucket_listv2_delimiter_whitespace
s3tests_boto3.functional.test_s3.test_bucket_list_delimiter_dot
s3tests_boto3.functional.test_s3.test_bucket_listv2_delimiter_dot
s3tests_boto3.functional.test_s3.test_bucket_list_delimiter_not_skip_special
s3tests_boto3.functional.test_s3.test_bucket_list_prefix_delimiter_alt
s3tests_boto3.functional.test_s3.test_bucket_listv2_prefix_delimiter_alt
```

The reason is that objects are pushed in the cache with full filename during upload. These infos are taken from the cache without correcting names according to delimiter and prefix in at https://github.com/nspcc-dev/neofs-s3-gw/blob/757600608c64c3a0b6eddfce8c5265a622cba2e5/api/layer/object.go#L535-L538 so it leads to invalid listing.

### PR DESCRIPTION
**Pull request by alexvanin**
Closes #611

### CODE REVIEW

**Comment by alexvanin** [comment]
@KirillovDenis @masterSplinter01 both commits provide working solution. First one is trivial, second I like a bit more. There is no longer two `objectInfoFromMeta()` and `objInfoFromMeta()` functions. Cache stores **only** unprocessed object infos, we do process them when it is necessary (when prefix and delimeter are available, e.g. in listing). By processing I mean fixing type (isDir) and name.

**Comment by KirillovDenis** [comment]
It seems we can process name by using `data.NodeVesion` struct before trying to get objectinfo from cache or NeoFS. It can help to reduce unnecessary requests.

**Review by KirillovDenis — DISMISSED** [review]

File: api/layer/util_test.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ func Test_objectInfoFromMeta(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
-			info := objectInfoFromMeta(bkt, tc.object, tc.prefix, tc.delimiter)
+			info := processObjectInfoName(objectInfoFromMeta(bkt, tc.object), tc.prefix, tc.delimiter)
```
Comment (masterSplinter01) [inline api/layer/util_test.go:161]: It doesn't matter a lot, but it'd be more consistent to rename `func Test_objectInfoFromMeta`

**Review by masterSplinter01 — APPROVED** [review]
The second solution looks prettier

**Pull request merged**